compiler-core/src/analyse/name.rs

```diff
@@ -0,0 +1,116 @@
+use ecow::EcoString;
+
+use crate::ast::{ArgNames, SrcSpan};
+
+use super::{BadNameKind, Error};
+use heck::{ToSnakeCase, ToUpperCamelCase};
+
+type BadName = (SrcSpan, EcoString);
```

Comment:
This name isn't right, it's not a bad name. It's a correction or such. Let's use a struct for this.